src/js/providers/tracks-mixin.ts

```diff
@@ -504,7 +504,7 @@ const Tracks: TracksMixin = {
                 'default': false
             };
             this.addTextTracks([track]);
-            this.trigger('subtitlesTracks', { tracks: this._textTracks });
+            this.trigger(SUBTITLES_TRACKS, { tracks: this._textTracks as TextTrackLike[] });
```

Comment:
Really clean PR! QQ about this section: Why do we need to cast this._textTracks as TextTrackLike[] here? Shouldn't it already be?

Reply:
`_textTracks` is typed `TextTrackLike[] | null`.
`event.tracks` is always expected to be `TextTrackLike[]`.

`this.addTextTracks` makes sure it's not null, but TypeScript doesn't know that, so casting it rather than adding an assignment or type guard to maintain current functionality and avoid adding extra code.

Reply:
Thanks, that cleared it up!